ChallengeTrackerAppAngular/src/app/services/user.service.ts

```diff
@@ -18,6 +18,16 @@ export class UserService {
 
   constructor(private httpClient: HttpClient) { }
 
+  createNewUser(newUser: User): Observable<number> {
+    return this.httpClient.put<number>(this.baseUrl, newUser, httpOptions).pipe(
+      catchError(this.handleError)
+    )
+  }
+  userLogin(username: string | undefined, password: string | undefined): Observable<User> {
+    return this.httpClient.get<User>(this.baseUrl +'/userLogin?username=' + username +'&password=' + 
```

Comment:
Please don't pass user credentials as query parameters. You can pass them through Authorization header in the http request with basic auth, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

@PrasannaVelisetti	Reply…

Reply:
in progress